tripal/tripal
# Render HTML tags in page titles, e.g. organism scientifc name

# New Feature

### Issue #1872 

### Tripal Version: 4.x

## Description
  - [x] ~~`TripalTokenParser()` should strip outer &lt;p&gt;&lt;/p&gt; tags - These are added if a field is configured to one of the HTML filters and you want to use that value in the title. Maybe we want to include italics in a gene name and show that in the title. The value will always be stored in the database table with wrapping &lt;p&gt;&lt;/p&gt; tags around it.~~ This is not needed now that XSS::filter is used
  - [x] Add test for the above change
  - [x] The entity list should support HTML tags. Once implemented, the organism content pages can appear like this
![2024-05-20_tripalcontentlist](https://github.com/tripal/tripal/assets/8419404/5580e396-22b3-4dac-a825-ecea549af91f)

  - [x] Create a hook to preprocess the title field, and convert the string value into markup. This will allow any HTML tags to be rendered.
  - [x] Create a hook to preprocess the page title block, again convert any string value into markup.
  - [x] Use XSS::filter to sanitize everything 🪣🧽
  - [x] The default organism title template can now include italics `<em>[organism_genus] [organism_species]</em> [organism_infraspecific_type] <em>[organism_infraspecific_name]</em>`
  - [x] Create an admin form to allow configuring the allowed HTML tags
    An organism page should look like this now
![2024-05-20_allworking](https://github.com/tripal/tripal/assets/8419404/70734750-470f-4104-a6ac-aca67a217295)


## Testing?
Because the organism default title template now has HTML tags for italics, testing is simple
 1. Make sure tests on this branch had no errors.
 1. Create a new organism
 1. Both versions of the title should show italics
 1. The entity listing should show the title with italics
 1. For good measure, create a different content type to make sure a "normal" title still works as before.
 1. Finally, create a publication with bold and italics in the title. See that it displays correctly.
 1. Go to Manage -> Tripal -> Tripal Entity Settings and remove all tags and save. Rebuild cache and confirm no bold or italics now displays.
 1. Test form validation. In that same form, enter something not a tag like `123` - only characters and spaces should be allowed. Put in too many spaces, they should reduce to a single space only between tags.
 
![2024-05-20_pub](https://github.com/tripal/tripal/assets/8419404/0e959c42-bb70-4230-8716-2d3978644a5d)

 I note that currently &lt;strong&gt; tags don't seem to render as bold in the page title block 🙈
 However this is a theme issue as the tags are there (Olivero 10.2.7-dev (default theme))
`<h1 class="title page-title"><strong>Tripal 4</strong> and other hazardous adventures <em>in silico</em></h1>` 
The entire &lt;h1&gt; element is already rendered as "bold", thus we don't see a difference.
![2024-05-20_h1](https://github.com/tripal/tripal/assets/8419404/de64bf27-0355-46ab-83ee-5ae7a5a72398)
